tasks/Task.cpp

```diff
@@ -211,3 +152,116 @@ void Task::cleanupHook()
 {
     TaskBase::cleanupHook();
 }
+
+bool Task::setRange(double value)
+{
+    head.SetStopRange(value);
+    //check if the desired range was really set.
+    if (value != head.GetStopRange())
+    {
+        std::cout << "Required range is larger than allowed, the maximum value " <<
+                head.GetStopRange() << " was set." << std::endl;
+    }
+    return (sonar_bvt_sdk::TaskBase::setRange(head.GetStopRange()));
+}
+
+bool Task::setTvg_slope(double value)
+{
+    //check if the desired tvg is valid.
+    if (value < 0 ) value = 0;
+    if (value > 10) value = 10;
+    head.SetTVGSlope(value);
+
+    return (sonar_bvt_sdk::TaskBase::setTvg_slope(value));
+}
+
+bool Task::setGain(double value)
+{
+    // convert gain percent to dB value
+    double db = 49 * value + 1;
+    head.SetGainAdjustment(db);
+    return (sonar_bvt_sdk::TaskBase::setGain(value));
+}
+
+void Task::setFluidType(sonar_bvt_sdk::fluid_type::FluidType type)
+{
+    switch(type)
+    {
+        case sonar_bvt_sdk::fluid_type::Saltwater:
+            head.SetFluidType(BVTSDK::FluidType::Saltwater);
+            break;
+        case sonar_bvt_sdk::fluid_type::Freshwater:
+            head.SetFluidType(BVTSDK::FluidType::Freshwater);
+            break;
+        case sonar_bvt_sdk::fluid_type::Other:
+            head.SetFluidType(BVTSDK::FluidType::Other);
+            break;
+        default:
+            throw std::runtime_error("The fluid type could not be set");
+    }
+}
+
+void Task::setImageResolution(sonar_bvt_sdk::image_resolution::ImageResolution resolution)
+{
+    switch(resolution)
+    {
+        case sonar_bvt_sdk::image_resolution::Off:
+            head.SetImageRes(BVTSDK::ImageRes::Off);
+            break;
+        case sonar_bvt_sdk::image_resolution::Low:
+            head.SetImageRes(BVTSDK::ImageRes::Low);
+            break;
+        case sonar_bvt_sdk::image_resolution::Medium:
+            head.SetImageRes(BVTSDK::ImageRes::Medium);
+            break;
+        case sonar_bvt_sdk::image_resolution::High:
+            head.SetImageRes(BVTSDK::ImageRes::High);
+            break;
+        case sonar_bvt_sdk::image_resolution::Auto:
+            head.SetImageRes(BVTSDK::ImageRes::Auto);
+            break;
+        default:
+            throw std::runtime_error("The image resolution could not be set");
```

Comment:
Better message: "Image resolution parameter does not match an enum value" (and throw std::invalid_argument)